.gitignore

```diff
@@ -1,12 +1,13 @@
+# shadow build dir
 /build
-/Debug
-.autotools
-.project
-.cproject
-.pydevproject
-.settings
-*.swp
+# area for local temp files
+/local
+# ide-specific settings
+.vscode
 cscope.*
 # ctags index
 /tags
 /src/tags
+# other temp files
+*.swp
+*Cargo.lock
```

Comment:
Can you replace this line with `/src/support/logger/rust_bindings/Cargo.lock`? The other `Cargo.lock` files should be checked into the repository.

@sporksmith Any thoughts about how we should handle the lock files? My understanding is they should be checked in to the repo for executable projects, and we should run `cargo update` every once in a while to update it. Do you agree, or is there anything else we should consider?

Reply:
FWIW, since the Cargo.toml already lists dependency versions (or version ranges), I'm fine with either approach: ignoring the Cargo.locks, or checking them in as Steven suggests.

Reply:
Based on https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries, checking in the Cargo.lock files probably makes sense. While we have some internal "libraries", we don't expect them to be used by projects outside of this repo.

> Can you replace this line with /src/support/logger/rust_bindings/Cargo.lock?

What's the reasoning for not checking in that `Cargo.lock`?

Reply:
My reasoning for not checking in the logging library's `Cargo.lock` is that the shadow_rs library's `Cargo.lock` already includes all of the pinned dependencies for the logging library. As far as I know the logging library's `Cargo.lock` isn't used for anything other than when building documentation, running `cargo check`, or when generating bindings.

Reply:
Done, thanks!